orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -165,6 +178,206 @@ def save_as_onnx(self, path):
+            other_export_options['training'] = torch.onnx.TrainingMode.TRAINING
+
+        torch.onnx._export(model, tuple(sample_inputs), f,
+                        input_names=input_names,
+                        output_names=output_names,
+                        opset_version=opset_version,
+                        _retain_param_name=True,
+                        example_outputs=tuple(sample_outputs),
+                        do_constant_folding=False,
+                        **other_export_options)
+
+        onnx_model = onnx.load_model_from_string(f.getvalue())
+
+        # Remove 'model_.' prefix introduced by model wrapper for initializers.
+        replace_name_dict = {}
+        for n in onnx_model.graph.initializer:
+            if n.name.startswith('model_.'):
+                replace_name_dict[n.name] = n.name[len('model_.'):]
+                n.name = replace_name_dict[n.name]
+        for n in onnx_model.graph.node:
+            for i, name in enumerate(n.input):
+                if name in replace_name_dict:
+                    n.input[i] = replace_name_dict[name]
+
+        # onnx model initializer may contain non-trainable registered buffers that are not part
+        # of pytorch model named parameteres.
+        named_parameters = model.model_.named_parameters() if hasattr(model, 'model_') else model.named_parameters()
+        assert set([n for n, t in named_parameters]).issubset(
+            set([n.name for n in onnx_model.graph.initializer])), \
+            "Initializer names do not match between PyTorch model and ONNX model, " \
+            "please report a bug to ONNX Runtime."
+
+        return onnx_model
+
+    def _init_onnx_model(self, inputs):
+            if self._onnx_model is not None:
+                return
+
+            if self._torch_model is not None:
+                # NOTE: pt model is moved to cpu to conserve gpu memory.
+                self._torch_model.cpu()
+                # torch buffers created using 'register_buffer' are not meant to be trainable.
+                torch_buffers = list(dict(self._torch_model.named_buffers()).keys())
+                #self.frozen_weights_ = self.frozen_weights_ + torch_buffers
+                self._onnx_model = self.convert_torch_model_loss_fn_to_onnx(torch.device('cpu'), inputs)
+
+            self._init_session()
+
+    def _prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, internal_loss_scale, *args, **kwargs):
+            fetches = None
+            """
+            if type(args) == tuple and len(args) == 1 and type(args[0]) == list:
+                input = tuple(args[0])
+            else:
+                input = args
+            """
+            input = args #added
+
+            for input_desc in input_desc_with_:
+                if input_desc[0] in kwargs:
+                    input = input + (kwargs[input_desc[0]],)
+            if internal_learning_rate is not None:
```

Comment:
345-348 belongs to another PR.

345: is for train_step
347: is for loss scaling